### PR TITLE
Feature: adding service for order status history

### PR DESCRIPTION
**Pull request by eapg**
* Adding service for manage order status histories in the Java etl impl.
* Adding repository for order status history

### CODE REVIEW

File: src/main/java/com/ea/restaurant/repository/OrderStatusHistoryRepository.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+                               WHERE osh.order_id IN :order_ids AND osh.entity_status = 'ACTIVE'
+                               GROUP BY osh.order_id
+                            )
+                                SELECT osh.*
+                                  FROM order_status_histories AS osh
+                            INNER JOIN latest_order_status_histories_cte loshcte
+                                    ON loshcte.order_id = osh.order_id AND loshcte.max_from_time = osh.from_time
+                            """,
+      nativeQuery = true)
+  List<OrderStatusHistory> findLastOrderStatusHistoriesByOrderIds(
```
Comment (reynoldmorel) [inline src/main/java/com/ea/restaurant/repository/OrderStatusHistoryRepository.java:27]: This should return one record right?

File: src/main/java/com/ea/restaurant/service/OrderStatusHistoryService.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.ea.restaurant.service;
+
+import com.ea.restaurant.entities.OrderStatusHistory;
+import java.util.List;
+
+public interface OrderStatusHistoryService {
+  List<OrderStatusHistory> findLastOrderStatusHistoriesByOrderIds(List<Long> orderIds);
+
+  void InsertNewOrUpdatedBatchOrderStatusHistories(List<OrderStatusHistory> orderStatusHistories);
```
Comment (reynoldmorel) [inline src/main/java/com/ea/restaurant/service/OrderStatusHistoryService.java:9]: Rename to `create`

File: src/main/java/com/ea/restaurant/repository/OrderStatusHistoryRepository.java (hunk 1 above)
Comment (eapg) [reply src/main/java/com/ea/restaurant/repository/OrderStatusHistoryRepository.java:27]: No. this method return a list of order status histories.

File: src/main/java/com/ea/restaurant/service/OrderStatusHistoryService.java (hunk 2 above)
Comment (eapg) [reply src/main/java/com/ea/restaurant/service/OrderStatusHistoryService.java:9]: No, this method insert order status histories after update, it is not a create method.

File: src/test/java/com/ea/restaurant/test/fixture/OrderStatusHistoryFixture.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  public static final OrderStatus TO_STATUS = OrderStatus.IN_PROCESS;
+  public static final String MONGODB_ORDER_STATUS_HISTORY_UUID = "MONGOUUID";
+  public static final EtlStatus ETL_STATUS = EtlStatus.PROCESSED;
+  public static final Long CREATED_BY = 1L;
+  public static final Instant CREATED_DATE = Instant.EPOCH;
+  public static final Long UPDATED_BY = 1L;
+  public static final Instant UPDATED_DATE = Instant.EPOCH;
+  public static final Status ENTITY_STATUS = Status.ACTIVE;
+
+  public static OrderStatusHistory buildOrderStatusHistory(OrderStatusHistory orderStatusHistory) {
```
Comment (reynoldmorel) [inline src/test/java/com/ea/restaurant/test/fixture/OrderStatusHistoryFixture.java:24]: Rename to `orderStatusHistoryExample`

Comment (reynoldmorel) [reply src/test/java/com/ea/restaurant/test/fixture/OrderStatusHistoryFixture.java:24]: Create a new `OrderStatusHistory` remember this is a fixture and a new object should be built from here. Fiuxtures must avoid side effects, so do not modify an instance, instead create a new one

File: src/test/java/com/ea/restaurant/service/impl/OrderStatusHistoryServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  @AfterEach
+  public void afterEach() throws Exception {
+    this.mockedAnnotations.close();
+  }
+
+  @Test
+  public void findLastOrderStatusHistoriesByOrderIds() {
+    var orderStatus1ToBeSet = new OrderStatusHistory(1L);
```
Comment (reynoldmorel) [inline src/test/java/com/ea/restaurant/service/impl/OrderStatusHistoryServiceImplTest.java:35]: `var orderStatus1ToBeSet = OrderStatusHistoryFixture.buildOrderStatusHistory(1L)`

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/test/java/com/ea/restaurant/test/fixture/OrderStatusHistoryFixture.java (hunk 3 above)
Comment (eapg) [reply src/test/java/com/ea/restaurant/test/fixture/OrderStatusHistoryFixture.java:24]: Done.

File: src/test/java/com/ea/restaurant/service/impl/OrderStatusHistoryServiceImplTest.java (hunk 4 above)
Comment (eapg) [reply src/test/java/com/ea/restaurant/service/impl/OrderStatusHistoryServiceImplTest.java]: Done.

**Review by reynoldmorel — APPROVED** [review]

**Pull request merged**